### DISCUSSION TITLE
Arduino ESP32 Bluetooth HiFi System with VU-Meter Using LED Strip

### DISCUSSION THREAD

**Discussion by Knottis** [pschatzmann/arduino-audio-tools#859] [Q&A] [answered]

**Comment by pschatzmann** [comment, marked as answer]
A couple of comments 
- If you just need to determine the volume, the VolumeOutput would be a more efficient choice. You could even get the volume of the left and right channel separately!
- You can also access the data stream before the volume has been applied: For further details check the class documentation of the ESP32-A2DP project:  BluetoothA2DPSink::set_raw_stream_reader().

**Comment by pschatzmann** [comment]
You never explained what you try to achieve: If you just want to measure the volume this is a pretty crazy way to do this and FFT is not the right tool.
